verifiers/internal/gcb/slsaprovenance/v1.0/provenance.go

```diff
@@ -0,0 +1,198 @@
+package v10
+
+// NOTE: Copy of github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.1
+// This holds an internal copy of in-toto-golang's structs for
+// SLSA predicates to handle GCB's incompatibility with the
+// published specification.
+// Specifically, GCB provenance currently produces a string for
+// ProvenancePredicate.Recipe.DefinedInMaterial rather than the SLSA compliant
+// signed integer. Because of this, we comment out the field and do not unmarshal
+// this in the Go struct. When comparing the envelope with the human-readable
+// content, this field is ignored!
+// GCB will later add compliant fields in the signed envelope, but NOT in the
+// human-readable component. Either disregard comparison between human-readable
+// summary and the signed envelope, or use this struct in comparison.
+
+import (
+	"encoding/json"
+	"fmt"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	intotov1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	serrors "github.com/slsa-framework/slsa-verifier/v2/errors"
+	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gcb/slsaprovenance/common"
+	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gcb/slsaprovenance/iface"
+)
+
+const (
+	// PredicateSLSAProvenance represents a build provenance for an artifact.
+	PredicateSLSAProvenance = intotov1.PredicateSLSAProvenance
+	BuildType               = "https://cloud.google.com/build/gcb-buildtypes/google-worker/v1"
+)
+
+var BuilderIDs = []string{
+	"https://cloudbuild.googleapis.com/GoogleHostedWorker",
+}
+
+// ProvenancePredicate is the provenance predicate definition.
+type ProvenancePredicate intotov1.ProvenancePredicate
+
+// GCBIntotoTextStatement if for code compatibility with v0.1 code.
+type GCBIntotoTextStatement Provenance
+
+// Provenance is GCB provenance.
+type Provenance struct {
+	intoto.StatementHeader
+	Pred ProvenancePredicate `json:"predicate"`
+}
+
+func New(payload []byte) (iface.Provenance, error) {
+	var provenance Provenance
+	if err := json.Unmarshal(payload, &provenance); err != nil {
+		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, err.Error())
```

Comment:
We can wrap multiple errors since Go 1.20
```suggestion
		return nil, fmt.Errorf("%w: %w", serrors.ErrorInvalidDssePayload, err)
```

Reply:
nice! Created https://github.com/slsa-framework/slsa-verifier/issues/692 for tracking